src/volleyball.js

```diff
@@ -18,7 +19,7 @@ function Volleyball(config = {}) {
 		// items shared between the request and response of just one cycle
 		const cycle = {
 			log: log,
-			id: makeId(),
+			id: config.makeId ? config.makeId() : makeId(),
```

Comment:
The compiler may be able to optimize this, but on principle I'd prefer init-time branching – add the following ca. line 17 (before defining `function volleyball`):

```js
const idGenerator = config.makeId || makeId
```

Then `cycle` would be defined as:

```js
const cycle = {
    log: log,
    id: idGenerator(),
    time: process.hrtime()
}
```

It may be overkill, but that's how I'd do it (similar to how I instantiate the `log` function above).

Reply:
Follow-up, I would also prefer to add a modicum of dynamic type-checking to this config.

```js
if ((makeId in config) && (typeof config.makeId) !== 'function') {
    throw TypeError('Volleyball config option `makeId` must be a function')
}
```

(untested)